torch/lib/c10d/ProcessGroupMPI.cpp

```diff
@@ -94,13 +94,21 @@ void checkSameSizeAndType(
 
 } // namespace
 
+std::vector<at::Tensor> ProcessGroupMPI::WorkMPI::result() {
+  if (outputs_) {
+    return *outputs_;
```

Comment:
I don't understand this. Since we are making a copy of the vector anyway, why do we need shared pointer?